### PR TITLE
Remove children of Document even if doc not in mirror

### PR DESCRIPTION
**Pull request by Juice10**
Be more explicit about identifying if something is a Document, and deleting its child elements

### CODE REVIEW

File: packages/rrdom/test/diff.test.ts [hunk 1]
```diff
@@ -1056,6 +1056,24 @@ describe('diff algorithm for rrdom', () => {
...
+      rrDocument.mirror.add(docType, getDefaultSN(docType, 1));
+      rrDocument.appendChild(docType);
+      const htmlEl = rrDocument.createElement('html');
+      rrDocument.mirror.add(htmlEl, getDefaultSN(htmlEl, 2));
+      rrDocument.appendChild(htmlEl);
+
+      diff(document, rrDocument, replayer);
+      expect(document.childNodes.length).toBe(2);
+      const element = document.childNodes[0] as HTMLElement;
+      expect(element.DOCUMENT_TYPE_NODE).toBe(10);
```
Comment (Mark-Fenng) [inline packages/rrdom/test/diff.test.ts:1074]: Is it supposed to be this?
```suggestion
      expect(element.nodeType).toBe(element.DOCUMENT_TYPE_NODE);
```
Cause the current assertion could be a tautology.

**Review by Mark-Fenng — COMMENTED** [review]
It's a great catch and I really love it!

**Review by Mark-Fenng — APPROVED** [review]

**Comment by Yuyz0112** [comment]
can be merge after resolving conlicts

**Comment by Juice10** [comment]
@Yuyz0112 done!

**Pull request merged**